tests/test_asyncio/test_connection.py

```diff
@@ -254,9 +254,8 @@ async def do_close():
     async def do_read():
         return await conn.read_response()
 
-    reader = mock.AsyncMock()
-    writer = mock.AsyncMock()
-    writer.transport = mock.Mock()
+    reader = mock.Mock(spec=asyncio.StreamReader)
```

Comment:
Teach we swami? Why is this pattern better than AsyncMock?

Reply:
I think we can keep the AsyncMock the key here is the "spec".    When a spec is used, they become synonymous.
The use of 'spec' is essential when mocking classes with some methods sync, ad some async, since it will create the 
correct mock methods.

Reply:
Oh, this went away when I rebased.  But yes, when using "spec", to mock a class instance, it will create async/sync methods as appropritate.  Here is the salient text:

> Setting the spec of a [Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock), [MagicMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock), or [AsyncMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.AsyncMock) to a class with asynchronous and synchronous functions will automatically detect the synchronous functions and set them as [MagicMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) (if the parent mock is [AsyncMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.AsyncMock) or [MagicMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock)) or [Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) (if the parent mock is [Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock)). All asynchronous functions will be [AsyncMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.AsyncMock).